plugins/modules/monitoring/newrelic_deployment.py

```diff
@@ -99,47 +102,61 @@ def main():
             application_id=dict(required=False),
             changelog=dict(required=False),
             description=dict(required=False),
-            revision=dict(required=False),
+            revision=dict(required=True),
             user=dict(required=False),
             appname=dict(required=False),
             environment=dict(required=False),
             validate_certs=dict(default=True, type='bool'),
         ),
         required_one_of=[['app_name', 'application_id']],
-        supports_check_mode=True
+        mutually_exclusive=[['app_name', 'application_id']],
     )
 
-    # build list of params
-    params = {}
-    if module.params["app_name"] and module.params["application_id"]:
-        module.fail_json(msg="only one of 'app_name' or 'application_id' can be set")
-
-    if module.params["app_name"]:
-        params["app_name"] = module.params["app_name"]
-    elif module.params["application_id"]:
-        params["application_id"] = module.params["application_id"]
+    if module.params['app_name']:
+        data = 'filter[name]=' + str(module.params['app_name'])
+        newrelic_api = 'https://api.newrelic.com/v2/applications.json'
+        headers = {'x-api-key': module.params['token'],
+                   'Content-Type': 'application/x-www-form-urlencoded'}
+        (resp, info) = fetch_url(module,
+                                 newrelic_api,
+                                 headers=headers,
+                                 data=data,
+                                 method='GET')
+        if info['status'] != 200:
+            module.fail_json(msg="unable to get application list from newrelic: %s" % info['msg'])
+        else:
+            body = json.loads(resp.read())
+        if body is None:
+            module.fail_json(msg='No Data for applications')
+        else:
+            app_id = body['applications'][0]['id']
+            if app_id is None:
+                module.fail_json(msg="App not found in\
```

Comment:
Please do not use `\` when not absolutely necessary. Lines can be up to 140 characters IIRC, so there's really no need to cut them that short.